aqua-ethos-setup/aqua_ethos.sh

```diff
@@ -123,6 +146,94 @@ function makePut {
+    if [ -n "$RESPONSE_LABEL" ]; then
+        log "Label \"$CREATE_LABEL\" already exists."
+    else
+        log "Label \"$CREATE_LABEL\" does not exist, creating it."
+        NEW_LABEL=$(makePost settings/labels "{ \"name\": \"$CREATE_LABEL\" }" | jq ".name")
+        if [ "$NEW_LABEL"="$CREATE_LABEL" ]; then
+            log "Label \"$CREATE_LABEL\" created successfully."
+        fi
+    fi
+
+}
+
+
+function tokenHaslabel {
+    TOKEN_NAME=$1
+    TOKEN_LABEL=$2
+    log "Checking whether token with name \"$TOKEN_NAME\" exists and has the label \"$TOKEN_LABEL\"."
+
+    JQ_FILTER+='.[]  | select([.logicalname == "'$TOKEN_NAME'"] | any)| select([.allowed_labels] | any) | select([.allowed_labels[] == "'$TOKEN_LABEL'"] | any) | .command.default '
+    BATCH_TOKEN_VALUE=$(makeGet hostsbatch body | jq "$JQ_FILTER")
+    if [[ -z "$BATCH_TOKEN_VALUE" ]]; then
+        log "Token \"$TOKEN_NAME\" does not exist with label \"$TOKEN_LABEL\"."
+        return 1
+    else
+        log "Token \"$TOKEN_NAME\" exists with label \"$TOKEN_LABEL\".  Command for agent installs is \"$BATCH_TOKEN_VALUE\"."
+        return 0
+    fi
+}
+
+function addBatch_install_token_with_label {
+
+    ADD_TOKEN_NAME=$1
+    ADD_TOKEN_VALUE=$2
+    SET_TOKEN_LABEL=$3
+
+    getGatewayid
+
+    createLabel_if_does_not_exist "$SET_TOKEN_LABEL"
+
+    HOSTBATCH_RULE=$(makeGet hostsbatch body | jq ".[] |.command|.default"|grep "production-token-value" && echo "200")
+    HOSTBATCH_RULE_FINAL=$(echo "${HOSTBATCH_RULE:(-3)}")
+
+    if !(tokenHaslabel "$ADD_TOKEN_NAME" "$SET_TOKEN_LABEL"); then
+        log "Creating token named \"$ADD_TOKEN_NAME\" with label \"$SET_TOKEN_LABEL\" and token \"$ADD_TOKEN_VALUE\"."
+        TOKEN_PAYLOAD='{"logicalname":"'$ADD_TOKEN_NAME'","token":"'$ADD_TOKEN_VALUE'","description":"Batch install for production hosts.","enforce":true,"allowed_labels":["'$SET_TOKEN_LABEL'"],"allowed_registries":["Docker Hub"],"gateways":"'$GATEWAY'"}'
+        log "$TOKEN_PAYLOAD"
+        BATCH_TOKEN_RESPONSE=$(makePost hostbatch "$TOKEN_PAYLOAD")
+        log "$BATCH_TOKEN_RESPONSE"
+        log "Validating that \"$ADD_TOKEN_NAME\" has been created."
+        if !(tokenHaslabel "$ADD_TOKEN_NAME" "$SET_TOKEN_LABEL"); then
+            log "Token does not have matching label."
+            log "--- Fail ---"
+            exit 1
+        fi
+    fi
+
+}
+
+addBatch_install_token_with_label "production-token" "production-token-value" "production approved"
+
+#scan Admin containers
+
+function urlEncoderepo() {
+  imageFullname=$1
+  echo $imageFullname | sed -e 's|/|%2F|g'
+}
+
+#ARTIFACTORY_IMAGES="iam-role-proxy:1.4.0 aws-ecr-login:2.0.0"
+
+if [[ -z "$ARTIFACTORY_IMAGES" ]]; then
+    log "ARTIFACTORY_IMAGES environment variable not set. Exiting..."
+    exit 1
+else
+      for image in $ARTIFACTORY_IMAGES;do
+           makePost scanner/registry/adobe-artifactory/image/$(urlEncoderepo $image)/scan
```

Comment:
Should you also assign labels on this images so that they will be approved for running?